railties/lib/rails/tasks/statistics.rake

```diff
@@ -1,37 +1,15 @@
 # frozen_string_literal: true
+require "rails/code_statistics"
 
-# While global constants are bad, many 3rd party tools depend on this one (e.g
-# rspec-rails & cucumber-rails). So a deprecation warning is needed if we want
-# to remove it.
-STATS_DIRECTORIES ||= [
-  %w(Controllers        app/controllers),
-  %w(Helpers            app/helpers),
-  %w(Jobs               app/jobs),
-  %w(Models             app/models),
-  %w(Mailers            app/mailers),
-  %w(Mailboxes          app/mailboxes),
-  %w(Channels           app/channels),
-  %w(Views              app/views),
-  %w(JavaScripts        app/assets/javascripts),
-  %w(Stylesheets        app/assets/stylesheets),
-  %w(JavaScript         app/javascript),
-  %w(Libraries          lib/),
-  %w(APIs               app/apis),
-  %w(Controller\ tests  test/controllers),
-  %w(Helper\ tests      test/helpers),
-  %w(Job\ tests         test/jobs),
-  %w(Model\ tests       test/models),
-  %w(Mailer\ tests      test/mailers),
-  %w(Mailbox\ tests     test/mailboxes),
-  %w(Channel\ tests     test/channels),
-  %w(Integration\ tests test/integration),
-  %w(System\ tests      test/system),
-]
+STATS_DIRECTORIES = ActiveSupport::Deprecation::DeprecatedObjectProxy.new(
+  CodeStatistics::DIRECTORIES,
+  "STATS_DIRECTORIES is deprecated! Use CodeStatistics::DIRECTORIES instead.",
```

Comment:
```suggestion
  "`STATS_DIRECTORIES` is deprecated in favor of `CodeStatistics::DIRECTORIES` and will be removed in Rails 7.2. is deprecated!",
```

Probably not 7.2, but you get the idea

Reply:
As `STATS_DIRECTORIES` is used primarily by third parties, the user might not be able to make the required changes.
So I'm not sure which version should remove it.
All other usages of DeprecatedObjectProxy don't mention a version either:
https://github.com/rails/rails/blob/121e0ad015c74eaaca82c6c9a85f5b2e0672ca0b/activesupport/lib/active_support/log_subscriber.rb#L65-L66
https://github.com/rails/rails/blob/121e0ad015c74eaaca82c6c9a85f5b2e0672ca0b/activesupport/lib/active_support/deprecation/proxy_wrappers.rb#L42